### PR TITLE
[22.01] Only stream stdout/stderr on kubernetes

### PR DESCRIPTION
**Pull request by nsoranzo**
Fix https://github.com/galaxyproject/galaxy/issues/13637 .

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: lib/galaxy_test/api/test_jobs.py [hunk 1]
```diff
@@ -205,25 +205,50 @@ def test_index_multiple_states_filter(self, history_id):
...
+            "tool_stderr",
+            "tool_stdout",
+            "update_time",
+        )
+
+        self.dataset_populator.wait_for_job(job_id, assert_ok=True)
+        show_jobs_response = self.dataset_populator.get_job_details(job_id, full=True)
+        job_details = show_jobs_response.json()
+        assert job_details["job_stdout"] == ""
+        assert job_details["job_stderr"] == ""
```
Comment (mvdbeek) [inline lib/galaxy_test/api/test_jobs.py]: I'm not sure we should assert things here, that's up to the DRM.

File: lib/galaxy_test/api/test_jobs.py [hunk 2]
```diff
@@ -205,25 +205,50 @@ def test_index_multiple_states_filter(self, history_id):
...
+        assert job_details["tool_stdout"] == "The bool is not true\n"
+        assert job_details["tool_stderr"] == "The bool is very not true\n"
+        assert job_details["stdout"] == "The bool is not true\n"
+        assert job_details["stderr"] == "The bool is very not true\n"
```
Comment (mvdbeek) [inline lib/galaxy_test/api/test_jobs.py]: Same here

File: lib/galaxy_test/api/test_jobs.py (hunk 1 above)
Comment (nsoranzo) [reply lib/galaxy_test/api/test_jobs.py]: But we know that API tests run under the local runner, no?

Comment (mvdbeek) [reply lib/galaxy_test/api/test_jobs.py]: No, they can target external servers and should work anywhere. We can do this with the integration tests though.

Comment (nsoranzo) [reply lib/galaxy_test/api/test_jobs.py]: I can change it to assert that this doesn't contain "The bool is not true", though.

Comment (mvdbeek) [reply lib/galaxy_test/api/test_jobs.py]: Yes, that works ... maybe. I don't know, I think it would still be more appropriate as an integration test

Comment (nsoranzo) [reply lib/galaxy_test/api/test_jobs.py]: I've changed the asserts, hope those are fine.

File: lib/galaxy/jobs/command_factory.py [hunk 3]
```diff
@@ -277,7 +278,10 @@ def append_command(self, command, sep=';'):
     def append_commands(self, commands):
         self.append_command("; ".join(c for c in commands if c))
 
-    def capture_stdout_stderr(self, stdout_file, stderr_file):
+    def capture_stdout_stderr(self, stdout_file, stderr_file, stream_stdout_stderr=False):
+        if not stream_stdout_stderr:
+            self.append_command(f'> "{stdout_file}" 2> "{stderr_file}"', sep="")
```
Comment (nsoranzo) [inline lib/galaxy/jobs/command_factory.py]:
```suggestion
            self.append_command(f"> '{stdout_file}' 2> '{stderr_file}'", sep="")
```

File: test/unit/app/jobs/test_command_factory.py [hunk 4]
```diff
@@ -18,17 +18,15 @@
...
-trap 'rm -f "$__out" "$__err"' EXIT
-tee -a '../outputs/tool_stdout' < "$__out" &
-tee -a '../outputs/tool_stderr' < "$__err" >&2 & """
 
 
 class TestCommandFactory(TestCase):
 
+    stream_stdout_stderr = False
+    TEE_LOG = " "
+    CAPTURE_AND_REDIRECT = f'> "../outputs/tool_stdout" 2> "../outputs/tool_stderr"{RETURN_CODE_CAPTURE}'
```
Comment (nsoranzo) [inline test/unit/app/jobs/test_command_factory.py]:
```suggestion
    CAPTURE_AND_REDIRECT = f"> '../outputs/tool_stdout' 2> '../outputs/tool_stderr'{RETURN_CODE_CAPTURE}"
```

**Review by nsoranzo — COMMENTED** [review]
Thanks for finishing this!

**Comment by nsoranzo** [comment]
@mvdbeek If you approve, I can merge.

**Review by mvdbeek — APPROVED** [review]

**Pull request merged**